src/components/register.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react'; // * Importa React e o hook useState
+
+// * Define o componente Register que recebe uma prop onRegister
+const Register = ({ onRegister }) => {
+    // Define estados para email, senha e confirmação de senha
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+
+    //  * Função que lida com o envio do formulário
+    const handleSubmit = (e) => {
+        e.preventDefault(); // Previne o comportamento padrão do formulário
+        if (password !== confirmPassword) { // Verifica se as senhas coincidem
+            alert('Passwords do not match'); // Alerta se as senhas não coincidem
+            return; // Sai da função se as senhas não coincidem
+        }
+        onRegister({ email, password }); // Chama a função onRegister com email e senha
+    };
+
+    return (
+        // Formulário que chama handleSubmit ao ser enviado
+        <form onSubmit={handleSubmit}>
+            <h2>Register</h2> {/* Título do formulário */}
+            <label>
```

Comment:
Não falto indicar a que input essa label pertence? No html a gente usa o for para indicar a quem a label pertence, no react é o htmlFor

vejam esse link: https://developer.mozilla.org/pt-BR/docs/Learn/Forms/How_to_structure_a_web_form